src/theoretical.jl

```diff
@@ -64,6 +64,21 @@ Evaluate the variogram at points `x` and `y`.
 #------------------
 # IMPLEMENTATIONS
 #------------------
+"""
+    NuggetEfect(n)
+
+A pure nugget effect (random) variogram with nugget `n`
+"""
+@with_kw struct NuggetEffect{T,D} <: Variogram{T,D}
+  nugget::T = 0.0
+  distance::D = Euclidean()
+end
+
+(γ::NuggetEffect)(h) = (h > 0) * γ.nugget
+Base.range(::NuggetEffect{T,D}) where {T,D} = zero(T)
```

Comment:
I think we can hard code the range to be Float64. The nugget type may be different in exotic applications and so we would be better off with a separate type for the range. Can you please refactor this to `Base.range(::NuggetEffect) = 0.0` for now?

Reply:
Nevermind, I think all other models are assuming a single type T anyways. Please ignore my comment above.